test/functional/aws_shared_functions.go

```diff
@@ -184,37 +187,34 @@ func GetS3BlobStorageResourceIDs(ctx context.Context, client client.Client, rhmi
 	return foundResourceIDs, foundErrors
 }
 
-// creates a session to be used in getting an api instance for aws
+// CreateAWSSession creates a session to be used in getting an api instance for aws
 func CreateAWSSession(ctx context.Context, client client.Client) (*session.Session, error) {
-	//retrieve aws credentials for creating an aws session
-	awsSecretAccessKey, awsAccessKeyID, err := getAWSCredentials(ctx, client)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get AWS credentials : %w", err)
-	}
-
-	//retrieve aws region for creating an aws session
 	region, err := getAWSRegion(ctx, client)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get AWS cluster region : %w", err)
 	}
-
-	//create new session for aws api's
-	sess, err := createAWSSession(awsSecretAccessKey, awsAccessKeyID, region)
+	awsConfig := aws.Config{
+		Region: aws.String(region),
+	}
+	isSTS, err := sts.IsClusterSTS(ctx, client, logger.NewLogger())
 	if err != nil {
-		return nil, fmt.Errorf("failed to create session : %w", err)
+		return nil, err
 	}
-	return sess, nil
-}
-
-// createAWSSession returns a new session from aws
-func createAWSSession(awsAccessKeyID, awsSecretAccessKey, region string) (*session.Session, error) {
-	sess, err := session.NewSession(&aws.Config{
-		Credentials: credentials.NewStaticCredentials(awsAccessKeyID, awsSecretAccessKey, ""),
-		Region:      aws.String(region),
-	})
+	if isSTS {
+		roleARN, err := sts.GetStsRoleArn(ctx, client, common.RHMIOperatorNamespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get role ARN: %w", err)
+		}
+		sess := session.Must(session.NewSession(&awsConfig))
+		awsConfig.Credentials = stscreds.NewCredentials(sess, roleARN)
+		return sess, nil
+	}
```

Comment:
While the above would work when running locally, I dont think it would work when the tests is run in a pod / cluster which would need to use the `NewWebIdentityRoleProvider` :thinking: 

* https://github.com/integr8ly/cloud-resource-operator/blob/270b698ed97a0072e134b160a169068a3f858c13/pkg/providers/aws/config.go#L172-L183

Also similar mounting of service account token should be done to delorean when creating the job spec I think :thinking: 

* https://github.com/integr8ly/cloud-resource-operator/blob/270b698ed97a0072e134b160a169068a3f858c13/config/manager/manager.yaml#L36-L45 
* https://github.com/integr8ly/delorean/blob/master/cmd/productTests.go#L233-L293